x-pack/legacy/plugins/siem/public/components/timeline/body/events/stateful_event.tsx

```diff
@@ -106,66 +100,84 @@ export const EmptyRow = styled.div`
   margin: 5px 0 5px 0;
 `;
 
-export class StatefulEvent extends React.Component<Props, State> {
-  private _isMounted: boolean = false;
+export const StatefulEvent = React.memo<Props>(
+  ({
+    actionsColumnWidth,
+    addNoteToEvent,
+    browserFields,
+    columnHeaders,
+    columnRenderers,
+    event,
+    eventIdToNoteIds,
+    getNotesByIds,
+    isEventViewer = false,
+    maxDelay = 0,
+    onColumnResized,
+    onPinEvent,
+    onUnPinEvent,
+    onUpdateColumns,
+    pinnedEventIds,
+    rowRenderers,
+    timelineId,
+    toggleColumn,
+    updateNote,
+  }) => {
+    const [expanded, setExpanded] = useState<{ [eventId: string]: boolean }>({});
+    const [initialRender, setInitialRender] = useState<boolean>(false);
+    const [showNotes, setShowNotes] = useState<{ [eventId: string]: boolean }>({});
 
-  public readonly state: State = {
-    expanded: {},
-    showNotes: {},
-    initialRender: false,
-  };
+    const divElement = useRef<HTMLDivElement | null>(null);
 
-  public divElement: HTMLDivElement | null = null;
+    const _isMounted = useRef<boolean>(false);
```

Comment:
I do not think that we need a useRef here for `_isMounted` we can avoid it like that
```
useEffect(() => {
      let _isMounted = true;

      requestIdleCallbackViaScheduler(
        () => {
          if (!initialRender && _isMounted) {
            setInitialRender(true);
          }
        },
        { timeout: maxDelay }
      );
      return () => {
        _isMounted = false;
      };
    }, []);
```